opm/output/eclipse/Summary.cpp

```diff
@@ -286,6 +287,32 @@ quantity rpr(const fn_args& args) {
     return { RPR , measure::pressure };
 }
 
+quantity rwip(const fn_args& args) {
+    const EclipseState* es = args.ecl_state;
+    if (es == nullptr)
+        return { 0.0 , measure::volume };
+
+    const auto& pair = args.regionCells.find( args.num );
+    if (pair == args.regionCells.end())
+        return { 0.0 , measure::volume };
+
+
+    double RWIP = 0;
+    const std::vector<double>& swat = args.state[data::Solution::key::SWAT];
+    const auto& cells = pair->second;
+    const auto& properties = args.ecl_state->get3DProperties();
+    const auto& porv_data = properties.getDoubleGridProperty("PORV").getData();
+
+    for (auto cell_index : cells) {
+	size_t global_index = args.grid.getGlobalIndex( cell_index );
+        RWIP += swat[cell_index] * porv_data[global_index];
```

Comment:
you probably also have to multiply this with B_W. (at least if RWIP is given in terms of surface volume, which IMO is the only sensible option.)

Also, I assume that "swat" corresponds to the water saturation given by the solution, not to the values given by the SWAT ECL property. (there is potential for confusion.)